pkg/worker/pool.go

```diff
@@ -12,6 +12,7 @@ import (
 type WorkerPool struct {
```

Comment:
Coding up a worker pool is something everyone eventually does in Go.

This one looks pretty good, but there are lots of ways to get this wrong. I would plan on coming back to this several times for bug fixes and/or refactoring. When you are "finished", I think this will be about 25% as much code as you have here (and in `worker.go`).

Reply:
Yeah, this is another one of those legacy packages that could maybe be better served by a third-party dependency. It used to do a fair bit of domain specific logic, but I've managed to remove that in this branch and prior work.

Not sure I understand this, could you elaborate?
> When you are "finished", I think this will be about 25% as much code as you have here (and in worker.go).